nipype/pipeline/plugins/multiproc.py

```diff
@@ -134,207 +118,174 @@ def __init__(self, plugin_args=None):
-            if node.inputs.terminal_output == 'stream':
-                node.inputs.terminal_output = 'allatonce'
-
-        self._task_obj[self._taskid] = \
-            self.pool.apply_async(run_node,
-                                  (node, updatehash, self._taskid),
-                                  callback=self._async_callback)
+        if getattr(node.inputs, 'terminal_output', '') == 'stream':
+            node.inputs.terminal_output = 'allatonce'
+
+        self._task_obj[self._taskid] = self.pool.apply_async(
+            run_node, (node, updatehash, self._taskid),
+            callback=self._async_callback)
         return self._taskid
 
     def _close(self):
         self.pool.close()
         return True
 
+    def _prerun_check(self, graph):
+        tasks_mem_gb = []
+        tasks_num_th = []
+        for node in graph.nodes():
+            tasks_mem_gb.append(node.mem_gb)
+            tasks_num_th.append(node.n_procs)
+
+        if np.any(np.array(tasks_mem_gb) > self.memory_gb):
+            logger.warning(
+                'Some nodes exceed the total amount of memory available '
+                '(%0.2fGB).', self.memory_gb)
+            if self.raise_insufficient:
+                raise RuntimeError('Insufficient resources available for job')
+
+        if np.any(np.array(tasks_num_th) > self.processors):
+            logger.warning(
+                'Some nodes demand for more threads than available (%d).',
+                self.processors)
+            if self.raise_insufficient:
+                raise RuntimeError('Insufficient resources available for job')
+
+    def _check_resources(self, running_tasks):
+        """
+        Make sure there are resources available
+        """
+        free_memory_gb = self.memory_gb
+        free_processors = self.processors
+        for _, jobid in running_tasks:
+            free_memory_gb -= min(self.procs[jobid].mem_gb, free_memory_gb)
+            free_processors -= min(self.procs[jobid].n_procs, free_processors)
+
+        return free_memory_gb, free_processors
+
     def _send_procs_to_workers(self, updatehash=False, graph=None):
-        """ Sends jobs to workers when system resources are available.
-            Check memory (gb) and cores usage before running jobs.
         """
-        executing_now = []
+        Sends jobs to workers when system resources are available.
+        """
 
-        # Check to see if a job is available
-        currently_running_jobids = np.flatnonzero((self.proc_pending == True) & \
-                                (self.depidx.sum(axis=0) == 0).__array__())
+        # Check to see if a job is available (jobs without dependencies not run)
+        # See https://github.com/nipy/nipype/pull/2200#discussion_r141605722
+        jobids = np.nonzero(~self.proc_done & (self.depidx.sum(0) == 0))[1]
+        jobids = np.flatnonzero(
+            ~self.proc_done & (self.depidx.sum(axis=0) == 0).__array__())
```

Comment:
is this intended to be chosen from? (like a or b)?

Reply:
Good catch. No, the second line shouldn't be there